### PR TITLE
fix(TFD-3689+3398): align kafka dataset CSV delimiter property with kinesis

### PR DESCRIPTION
**Pull request by ftom**
**What is the current behavior?** (You can also link to an open issue here)
Kafka uses a single string property whereas Kinesis offers a preset enum + string property
Kafka does not consider the Avro schema required when selecting Avro format
**What is the new behavior?**
Kafka uses the same layout for CSV delim
Avro schema is now a required field when selecting Avro format

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1803/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1803/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1804/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1804/)

File: components/components-kafka/kafka-definition/src/main/java/org/talend/components/kafka/dataset/KafkaDatasetProperties.java [hunk 1]
```diff
@@ -46,12 +46,15 @@
...
+            PropertyFactory.newEnum("fieldDelimiter", FieldDelimiterType.class).setValue(FieldDelimiterType.SEMICOLON);
+
+    public Property<String> specificFieldDelimiter = PropertyFactory.newString("specificFieldDelimiter", ";");
 
     // Property for avro
-    public Property<Boolean> isHierarchy = PropertyFactory.newBoolean("isHierarchy", false);
+    public Property<Boolean> isHierarchy = PropertyFactory.newBoolean("isHierarchy", true);
 
-    public Property<String> avroSchema = PropertyFactory.newString("avroSchema");
+    public Property<String> avroSchema = PropertyFactory.newString("avroSchema").setRequired();
```
Comment (RyanSkraba) [inline components/components-kafka/kafka-definition/src/main/java/org/talend/components/kafka/dataset/KafkaDatasetProperties.java]: Can you check this when running?  avroSchema is only required when the Avro format is used, and there doesn't seem to be a default.  Can we change the required state in refreshForms?

File: components/components-kafka/kafka-definition/src/main/resources/org/talend/components/kafka/dataset/KafkaDatasetProperties.properties [hunk 2]
```diff
@@ -2,7 +2,12 @@ property.topic.displayName=Topic name
 property.valueFormat.displayName=Value format
 property.isHierarchy.displayName=Use custom avro format
 property.avroSchema.displayName=Custom avro schema
-property.fieldDelimiter.displayName=Field delimiter
+property.fieldDelimiter.displayName=Field Delimiter
```
Comment (RyanSkraba) [inline components/components-kafka/kafka-definition/src/main/resources/org/talend/components/kafka/dataset/KafkaDatasetProperties.properties]: I'm pretty sure UX has settled on "sentence case" for displayed text: `Field delimiter` and `Specific field delimiter`.

File: components/components-kafka/kafka-definition/src/main/resources/org/talend/components/kafka/dataset/KafkaDatasetProperties.properties [hunk 3]
```diff
@@ -2,7 +2,12 @@ property.topic.displayName=Topic name
...
+property.possiblevalue.SEMICOLON.displayName=Semicolon
+property.possiblevalue.COMMA.displayName=Comma
+property.possiblevalue.TABULATION.displayName=Tabulation
```
Comment (RyanSkraba) [inline components/components-kafka/kafka-definition/src/main/resources/org/talend/components/kafka/dataset/KafkaDatasetProperties.properties]: :/  Small -- nobody says Tabulation.

File: components/components-kafka/kafka-definition/src/main/java/org/talend/components/kafka/dataset/KafkaDatasetProperties.java (hunk 1 above)
Comment (ftom) [reply components/components-kafka/kafka-definition/src/main/java/org/talend/components/kafka/dataset/KafkaDatasetProperties.java]: fixed

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1813/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1813/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1814/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1814/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1815/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1815/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1816/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1816/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1818/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1818/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Afix%2FTFD-3689%2Falign-kafka-field-delim)

**Pull request merged**